perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_loa.java

```diff
@@ -0,0 +1,57 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.*;
+import cz.metacentrum.perun.core.api.exceptions.*;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributesModuleImplApi;
+
+import java.util.List;
+
+/**
+ *
+ * @author Pavel Vyskocil vyskocilpavel@muni.cz
+ */
+public class urn_perun_user_attribute_def_virt_loa extends UserVirtualAttributesModuleAbstract  implements UserVirtualAttributesModuleImplApi {
+
+	public void checkAttributeValue(PerunSessionImpl sess, User user, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		if(!attribute.equals(getAttributeValue(sess, user, attribute))) throw new WrongAttributeValueException(attribute, user, "Attribute value is not the highest value from user's UserExtSources Loas.");
+	}
+
+	public Attribute fillAttribute(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException, WrongAttributeAssignmentException {
+		return new Attribute(attributeDefinition);
+	}
+
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+
+		List<UserExtSource> extSources = sess.getPerunBl().getUsersManagerBl().getActiveUserExtSources(sess, user);
+		Integer maxLoa = 0;
+		for(UserExtSource e : extSources) {
+			if(maxLoa < e.getLoa()) maxLoa = e.getLoa();
+		}
+		Attribute attribute = new Attribute(attributeDefinition);
+		attribute.setValue(maxLoa.toString());
+		return attribute;
+	}
+
+	public boolean setAttributeValue(PerunSessionImpl sess, User user, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException {
+		//No need to set, its same like check for this module, and check is used after every set
+		return false;
+	}
+
+	public void removeAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		//Not suported yet.
+		throw new InternalErrorException("Can't remove value of this virtual attribute this way. " + attributeDefinition);
+	}
+
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_USER_ATTR_VIRT);
+		attr.setFriendlyName("loa");
+		attr.setDisplayName("Level of assurance");
+		attr.setType(Integer.class.getName());
```

Comment:
We historically use "String" value type for LoA, since it's like that in all production DBs. Please fix type to String in both modules (user|member:virt:loa). Thank you.